docs/spec.md

```diff
@@ -2633,6 +2633,52 @@ planning to address this in
 
 &nbsp;[More Examples](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/interpret/dot_general.mlir)
 
+### dynamic_iota
+
+#### Semantics
+
+Fills an `output` tensor with values in increasing order starting from zero
+along the `iota_dimension` dimension. `output` tensor shape is specified
+dynamically via `output_shape` input. More formally,
```

Comment:
@sdasgup3 Should we include formal calculations for dynamic ops? `result_index` is technically unknown until runtime. Maybe it's better to say something like this, which I borrowed from [StablehloOps.td](https://github.com/openxla/stablehlo/blob/205f405613a14554095d678ebd87c1ced924f7af/stablehlo/dialect/StablehloOps.td#L130-L132):
```
Informally, this operation does the same thing as IotaOp except that the
result shape is specified dynamically via `output_shape`:
https://github.com/openxla/stablehlo/blob/main/docs/spec.md#iota
```

Reply:
+1

Reply:
for now removed formal calculations. 

keeping this chat open

Reply:
From https://github.com/openxla/stablehlo/blob/a7904883754a455bfd0c7636cbac1f99c2c24fe5/docs/spec.md?plain=1#L6605, the argument of `index_space` can be a placeholder. 
```
Placeholder` (future values, e.g. `lhs`, `rhs` or `result`; their actual
values are not known yet, only their types are known)
```

That means we can use the dynamic information of `output_shape` to bounds of `result_index` and that is all we need for the formalism.

For example,
```

More formally, for all result_index in index_space(TensorType(output_shape, si64)),

output[result_index] = constant(is_quantized(output) ?
quantize(result_index[iota_dimension], element_type(output)) :
result_index[iota_dimension], element_type(output))

```

Reply:
Btw `TensorType` is not needed since it is also defined on `Value` (`output_shape` is a `Value`):
```
Fills a `result` tensor with values in increasing order starting from zero
along the `iota_dimension` dimension. `result` shape is provided
dynamically via `output_shape`.

More formally, for all `result_index` in `index_space(output_shape)`,

`output[result_index] = constant(is_quantized(output) ?
quantize(result_index[iota_dimension], element_type(output)) :
result_index[iota_dimension], element_type(output))`.
```

Reply:
Thanks folks, discussion was helpful to get more insight on formal calculations. Updated.